types/IUser.ts

```diff
@@ -0,0 +1,18 @@
+export default interface IUser {
+  id: string;
+  name: string;
+  givenName: string;
+  familyName: string;
+  email: string;
+  profilePicture: string;
+  bio?: string;
+  winStreak: number;
+  longestWinStreak: number;
+  lastWin?: Date;
+  gamesPlayed: number;
+  gamesWon: number;
+  guessDistribution: number[];
+  pastGuesses: (number[][] | null)[];
+  pastTimes: (number | null)[];
```

Comment:
maybe switching from number to time datatype will make it easier to handle during average time calculations and leaderboard calculations.

Reply:
There is no Time data type unfortunately, there is only a Date data type. The number represents the number of seconds spent on the puzzle, so it should still be easy to handle (convert to minutes by dividing by 60, calculating average should still be straightforward).

Reply:
ah I see, that sounds like a great method!